### PR TITLE
🔥 decorate a class 🔥 

### PR DESCRIPTION
**Pull request by roniy68**
# Highlights 🥇 

## What I Have Done: 
✅  Create a class Nameable.
✅  Turn Person class to Nameable
✅  Make sure that the Person class inherits from Nameable
✅  Make sure that this class has a method correct_name implemented and returns the name attribute.
✅ Prepare base Decorator
✅ Make sure that it inherits from Nameable.
✅ In the constructor I assign a nameable object from params to an instance variable.
✅ Implement the correct_name method that returns the result of the 
correct_name method of the @nameable.

### Changes details
***NOTE: please add @roniy68 to your comment so that I can get notified***
## Missed anything?
- [ ] Explained the purpose of this PR.
- [ ] Self reviewed the PR.
- [ ] Added or updated test cases.
- [ ] Informed of breaking changes, testing and migrations (if applicable).
- [ ] Updated documentation (if applicable).
- [ ] Attached screenshots (if applicable).

## Followed Rules: 
- Made sure [no linter errors](https://github.com/microverseinc/linters-config)
-  Followed [Github Flow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/github_flow.md)
-  Documented [professionaly](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md)
- Follow our list of [best practices for Ruby](https://github.com/microverseinc/curriculum-ruby/blob/main/articles/ruby_best_practices.md).

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+class TrimmerDecorator < BaseDecorator
+  def correct_name
+    if @nameable.correct_name.length > 10
+      @nameable.correct_name[0..9]
+    else
+      @nameable.correct_name
+    end
+  end
+end
```
Comment (prolajumokeoni) [inline person.rb]: - [x] It would be better to separate the three classes in this file to different files this will make your code more readable 👍🏿

**Review by prolajumokeoni — CHANGES_REQUESTED** [review]
Hi @roniy68 
Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️  The three classes created successfully 



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 1 above)
Comment (roniy68) [reply person.rb]: 😄 Though I think its not necessary to differentiate small classes into different files. But In this case I did approved your requirements. 
Thanks
Happy Coding 😄

**Review by TadesseDev — APPROVED** [review]
### Hi @roniy68 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
giphy (1)



## Highlights
 - [x] Good job making the changes requested by the previous code reviewer. ✅

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**